src/alloy/README.md

```diff
@@ -0,0 +1,37 @@
+# Grafana Alloy
+
+Grafana Alloy is a distribution of the OpenTelemetry (OTel) Collector. Within UDS Core it is primarily used for log collection and shipping to destinations (like Loki and S3).
+
+## Switching from Promtail to Alloy
+
+Within UDS Core we have made the decision to switch from Promtail (historically the log collector/shipper of choice) to Grafana Alloy. The below contains primary motivating factors and impacts of this choice.
+
+### Motivations
+
+Promtail has historically been the tool of choice for log collection/shipping when using Loki. It provides a very lightweight layer to scrape logs from pods and hosts, label them with additional metadata, and ship them to Loki.
+
+One of the main issues that has arisen with Promtail is its limited output/export options. Promtail only supports sending logs to one or more Loki instances. A common requirement in production environments is to ship logs to a secondary destination for collection/analysis by security teams and SIEM tools. Promtail is currently listed as [feature complete](https://grafana.com/docs/loki/latest/send-data/promtail/) so there is no expectation that additional export functionality would be added.
+
+### Goals and Options
+
+In choosing an alternative to Promtail we have a few primary objectives:
+1. Chosen tool must be capable of gathering host and pod logs: This has been our primary usage of Promtail in the past - gathering pods logs and host logs (to include k8s audit logs, controlplane logs, etc).
+1. Provide a tool that has a generic export option, as well as direct Loki integration: Generally we are looking for S3/S3-compatible object storage integrations as this is a common ask from end users. While specific SIEM tool integrations can be nice, it's more valuable to have something generic that most other tools can pick logs up from. In addition, a direct integration with Loki that makes it easy to label logs for indexing/querying in Loki is highly desirable.
+1. Choose a tool that does not require major changes in our logging stack, but is flexible for future adjustments to the stack: As we do have active users of our product we want to be careful in switching tools, so ideally we would like a tool that is a "drop-in" replacement. However, we don't want to rule out future changes to other pieces of the stack (i.e. Loki) so choosing a tool that doesn't lock us into Loki is important.
+1. Focus on the log collection/shipping problem: While there are a number of tools that offer far more than just logging pipelines (metrics, traces, etc), we don't currently see a need to focus on these tools. These features are seen as a nice to have, but not being evaluated as the focus here.
```

Comment:
Maybe not for this document but I am really curious how the metrics side of all of this could play out. Some questions:
* Do we need to export our metrics out of cluster as well? Is alloy a good fit there? 
* Is it conceivable we could drop other uds-core components?  Just glancing at alloy it supports metrics and surprising to me at least, things like ServiceMonitors?

Reply:
Yeah I intentionally left off metrics (and traces) from here. One key distinction/point to make - OTel Collector/Alloy is as the name suggests just a collector. My understanding is that ultimately you still need somewhere to store and something to query (ex: still need Loki + Grafana for log storage/querying). For metrics I believe the setup would end up looking like Alloy scraping metrics then pushing them to Prometheus, so you're really just ending up with an extra middleman.

Definitely could be evaluated, but in our current state I don't think it would reduce components/complexity and haven't heard any ask/need to export metrics. It does provide an option for that if we determine that is necessary in the future.